lib/link_checker.rb

```diff
@@ -0,0 +1,86 @@
+class LinkChecker
+  class << self
+    def extract_urls(data)
+      new(data).extract_urls
+    end
+
+    def link_checker_api_bearer_token
+      ENV["LINK_CHECKER_API_BEARER_TOKEN"]
+    end
+
+    def for_files_at(path)
+      path = File.join(path, "**/*.erb") unless path.to_s.include?("*")
+      data = Dir.glob(path).map { |filename| File.read(filename) }
+      new(data)
+    end
+  end
+
+  attr_reader :data
+
+  def initialize(data)
+    @data = data
+  end
+
+  def urls
+    # Gather urls in a Set to allow simple combination and duplicate removal
+    @urls ||= extract_urls.to_set
+  end
+
+  def extract_urls
+    case data
+    when String
+      uri = URI.extract(data, %w[http https])
```

Comment:
Does this pick only absolute paths? Do we want to check for relative paths too?

Reply:
[`URI.extract`](https://ruby-doc.org/stdlib-2.7.1/libdoc/uri/rdoc/URI/RFC2396_Parser.html#method-i-extract) retrieves full uri. This is testing the links out of the app. Relative paths can be easily tested within the app tests.